packages/xstate-lit/test/UseActor.ts

```diff
@@ -0,0 +1,71 @@
+import { html, LitElement } from 'lit';
+import type { Snapshot } from 'xstate';
+import { fromPromise } from 'xstate/actors';
+import { fetchMachine } from './fetchMachine.ts';
+import { UseMachine } from '../src/index.ts';
+
+const onFetch = () =>
+  new Promise<string>((res) => {
+    setTimeout(() => res('some data'), 50);
+  });
+
+const fMachine = fetchMachine.provide({
+  actors: {
+    fetchData: fromPromise(onFetch)
+  }
+});
+
+export class UseActor extends LitElement {
+  fetchController: UseMachine<typeof fetchMachine> = new UseMachine(this, {
+    machine: fMachine,
+    options: {
+      snapshot: this.persistedState
+    }
+  });
+
+  get persistedState(): Snapshot<any> | undefined {
+    return undefined;
+  }
+
+  override createRenderRoot() {
+    return this;
+  }
+
+  override render() {
+    return html`
+      <slot></slot>
+      <div>
+        ${this.fetchController.snapshot.matches('idle')
```

Comment:
I don't know the XState API, but could the control offer a `@lit/task`-like render helper that's a bit more declarative?

Something like:
```ts
this.fetchController.render({
  idle: () => html`...`,
  loading: () => html`...`,
  success: () => html`...`,
});
```

Could such an API be type safe wrt the the state names?

Reply:
Hmm, I'm not quite sure if those two concepts can be combined. We would need @davidkpiano  and  @Andarist 's input on this.

Reply:
Which two concepts?